packages/teams-js/src/internal/telemetry.ts

```diff
@@ -50,6 +56,60 @@ export const enum ApiVersionNumber {
   V_3 = 'v3',
 }
 
+/**
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export class HostToAppMessageDelayTelemetry {
+  private static callbackInformation: Map<MessageUUID, CallbackInformation> = new Map();
+
+  /**
+   * @internal
+   * Limited to Microsoft-internal use
+   *
+   * Store information about a particular message.
+   * @param messageUUID The message id for the request.
+   * @param callbackInformation The information of the callback.
+   */
+  public static storeCallbackInformation(messageUUID: MessageUUID, callbackInformation: CallbackInformation): void {
+    HostToAppMessageDelayTelemetry.callbackInformation.set(messageUUID, callbackInformation);
+  }
+
+  /**
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  public static clearMessages(): void {
```

Comment:
when is clear invoked?

Reply:
I agree! We can do this after the registered handler invocation. Maybe not clear all the messages but at least the one that was already handled by the function.